Flaky test: TempDir RemoveAll cleanup: [...] The process cannot access the file because it is being used by another process

Test flake on Windows:

https://github.com/pulumi/pulumi/actions/runs/14042855676/job/39317279602?pr=19009#step:41:212

```
=== FAIL: . TestLanguageImportSmoke/python (23.95s)
    environment.go:85: Created new test environment:  C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokepython2071463712\001
    environment.go:205: Running command pulumi login --cloud-url file://C:/Users/RUNNER~1/AppData/Local/Temp/TestLanguageImportSmokepython2071463712/001
    environment.go:205: Running command pulumi new python --yes
    environment.go:205: Running command pulumi import --yes random:index/randomId:RandomId identifier p-9hUg
    history_test.go:28: error cleaning up test directory "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestLanguageImportSmokepython2071463712\\001": remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokepython2071463712\001\project: The process cannot access the file because it is being used by another process.
    testing.go:1267: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokepython2071463712\001\project: The process cannot access the file because it is being used by another process.

=== FAIL: . TestLanguageImportSmoke/java (22.31s)
    environment.go:85: Created new test environment:  C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokejava3269083010\001
    environment.go:205: Running command pulumi login --cloud-url file://C:/Users/RUNNER~1/AppData/Local/Temp/TestLanguageImportSmokejava3269083010/001
    environment.go:205: Running command pulumi new java --yes
    environment.go:205: Running command pulumi import --yes random:index/randomId:RandomId identifier p-9hUg
    history_test.go:28: error cleaning up test directory "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestLanguageImportSmokejava3269083010\\001": remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokejava3269083010\001\project: The process cannot access the file because it is being used by another process.
    testing.go:1267: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokejava3269083010\001\project: The process cannot access the file because it is being used by another process.

=== FAIL: . TestLanguageImportSmoke/go (12.14s)
    environment.go:85: Created new test environment:  C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokego1417502556\001
    environment.go:205: Running command pulumi login --cloud-url file://C:/Users/RUNNER~1/AppData/Local/Temp/TestLanguageImportSmokego1417502556/001
    environment.go:205: Running command pulumi new go --yes
    environment.go:205: Running command pulumi import --yes random:index/randomId:RandomId identifier p-9hUg
    history_test.go:28: error cleaning up test directory "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestLanguageImportSmokego1417502556\\001": remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokego1417502556\001\project: The process cannot access the file because it is being used by another process.
    testing.go:1267: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokego1417502556\001\project: The process cannot access the file because it is being used by another process.

=== FAIL: . TestLanguageImportSmoke/yaml (5.60s)
    environment.go:85: Created new test environment:  C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokeyaml2636839738\001
    environment.go:205: Running command pulumi login --cloud-url file://C:/Users/RUNNER~1/AppData/Local/Temp/TestLanguageImportSmokeyaml2636839738/001
    environment.go:205: Running command pulumi new yaml --yes
    environment.go:205: Running command pulumi import --yes random:index/randomId:RandomId identifier p-9hUg
    history_test.go:28: error cleaning up test directory "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestLanguageImportSmokeyaml2636839738\\001": remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokeyaml2636839738\001\project: The process cannot access the file because it is being used by another process.
    testing.go:1267: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokeyaml2636839738\001\project: The process cannot access the file because it is being used by another process.

=== FAIL: . TestLanguageImportSmoke/nodejs (38.33s)
    environment.go:85: Created new test environment:  C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokenodejs3670192228\001
    environment.go:205: Running command pulumi login --cloud-url file://C:/Users/RUNNER~1/AppData/Local/Temp/TestLanguageImportSmokenodejs3670192228/001
    environment.go:205: Running command pulumi new typescript --yes
    environment.go:205: Running command pulumi import --yes random:index/randomId:RandomId identifier p-9hUg
    history_test.go:28: error cleaning up test directory "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestLanguageImportSmokenodejs3670192228\\001": remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokenodejs3670192228\001\project: The process cannot access the file because it is being used by another process.
    testing.go:1267: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokenodejs3670192228\001\project: The process cannot access the file because it is being used by another process.

=== FAIL: . TestLanguageImportSmoke/dotnet (15.84s)
    environment.go:85: Created new test environment:  C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokedotnet669506834\001
    environment.go:205: Running command pulumi login --cloud-url file://C:/Users/RUNNER~1/AppData/Local/Temp/TestLanguageImportSmokedotnet669506834/001
    environment.go:205: Running command pulumi new csharp --yes
    environment.go:205: Running command pulumi import --yes random:index/randomId:RandomId identifier p-9hUg
    history_test.go:28: error cleaning up test directory "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestLanguageImportSmokedotnet669506834\\001": remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokedotnet669506834\001\project: The process cannot access the file because it is being used by another process.
    testing.go:1267: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestLanguageImportSmokedotnet669506834\001\project: The process cannot access the file because it is being used by another process.

=== FAIL: . TestLanguageImportSmoke (118.17s)

=== FAIL: . TestImportVersionSmoke (10.95s)
    environment.go:85: Created new test environment:  C:\Users\RUNNER~1\AppData\Local\Temp\TestImportVersionSmoke665381590\001
    environment.go:205: Running command pulumi login --cloud-url file://C:/Users/RUNNER~1/AppData/Local/Temp/TestImportVersionSmoke665381590/001
    environment.go:205: Running command pulumi new D:\a\pulumi\pulumi\tests\testdata\random_go --yes
    environment.go:205: Running command pulumi plugin install resource random 4.16.7
    environment.go:205: Running command pulumi plugin install resource random 4.13.0
    environment.go:205: Running command pulumi import --yes random:index/randomId:RandomId identifier p-9hUg
    environment.go:205: Running command pulumi stack export
    history_test.go:28: error cleaning up test directory "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestImportVersionSmoke665381590\\001": remove C:\Users\RUNNER~1\AppData\Local\Temp\TestImportVersionSmoke665381590\001\project: The process cannot access the file because it is being used by another process.
    testing.go:1267: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestImportVersionSmoke665381590\001\project: The process cannot access the file because it is being used by another process.

=== FAIL: . TestImportParameterizedSmoke (44.51s)
    environment.go:85: Created new test environment:  C:\Users\RUNNER~1\AppData\Local\Temp\TestImportParameterizedSmoke4206137889\001
    environment.go:205: Running command pulumi login --cloud-url file://C:/Users/RUNNER~1/AppData/Local/Temp/TestImportParameterizedSmoke4206137889/001
    environment.go:205: Running command pulumi new D:\a\pulumi\pulumi\tests\testdata\random_python --yes
    environment.go:205: Running command pulumi plugin install resource terraform-provider 0.3.1
    environment.go:205: Running command pulumi package add terraform-provider hashicorp/random 3.6.3
    environment.go:205: Running command pulumi install
    environment.go:205: Running command pulumi up --yes
    environment.go:205: Running command pulumi import --yes random:index/id:Id identifier p-9hUg
    environment.go:205: Running command pulumi stack export
    history_test.go:28: error cleaning up test directory "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestImportParameterizedSmoke4206137889\\001": remove C:\Users\RUNNER~1\AppData\Local\Temp\TestImportParameterizedSmoke4206137889\001\project: The process cannot access the file because it is being used by another process.
    testing.go:1267: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestImportParameterizedSmoke4206137889\001\project: The process cannot access the file because it is being used by another process.

=== FAIL: . TestImportParameterizedSmokeFreshState (37.14s)
    environment.go:85: Created new test environment:  C:\Users\RUNNER~1\AppData\Local\Temp\TestImportParameterizedSmokeFreshState875474990\001
    environment.go:205: Running command pulumi login --cloud-url file://C:/Users/RUNNER~1/AppData/Local/Temp/TestImportParameterizedSmokeFreshState875474990/001
    environment.go:205: Running command pulumi new D:\a\pulumi\pulumi\tests\testdata\random_python --yes
    environment.go:205: Running command pulumi plugin install resource terraform-provider 0.3.1
    environment.go:205: Running command pulumi package add terraform-provider hashicorp/random 3.6.3
    environment.go:205: Running command pulumi install
    environment.go:205: Running command pulumi import --yes random:index/id:Id identifier p-9hUg
    environment.go:205: Running command pulumi stack export
    history_test.go:28: error cleaning up test directory "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestImportParameterizedSmokeFreshState875474990\\001": remove C:\Users\RUNNER~1\AppData\Local\Temp\TestImportParameterizedSmokeFreshState875474990\001\project: The process cannot access the file because it is being used by another process.
    testing.go:1267: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestImportParameterizedSmokeFreshState875474990\001\project: The process cannot access the file because it is being used by another process.
```

--- a/sdk/go/common/testing/environment.go
+++ b/sdk/go/common/testing/environment.go
@@ -76,11 +76,15 @@ func WriteYarnRCForTest(root string) error {
 
 // NewEnvironment returns a new Environment object, located in a temp directory.
 func NewEnvironment(t *testing.T) *Environment {
-	root := t.TempDir()
+	//nolint:usetesting // We control the lifecycle of the environment.
+	root, err := os.MkdirTemp("", "test-env")
+	assert.NoError(t, err, "creating temp directory")
 	assert.NoError(t, WriteYarnRCForTest(root), "writing .yarnrc file")
 
 	// We always use a clean PULUMI_HOME for each environment to avoid any potential conflicts with plugins or config.
-	home := t.TempDir()
+	//nolint:usetesting // We control the lifecycle of the environment.
+	home, err := os.MkdirTemp("", "test-env-home")
+	assert.NoError(t, err, "creating temp PULUMI_HOME directory")
 
 	t.Logf("Created new test environment:  %v", root)
 	return &Environment{

--- a/sdk/go/common/testing/environment.go
+++ b/sdk/go/common/testing/environment.go
@@ -16,6 +16,7 @@ package testing
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
 	"io"
 	"os"
@@ -119,26 +120,29 @@ func (e *Environment) ImportDirectory(path string) {
 	}
 }
 
-// DeleteEnvironment deletes the environment's RootPath, and everything underneath it.
+// DeleteEnvironment deletes the environment's HomePath and RootPath, and everything underneath them.
 func (e *Environment) DeleteEnvironment() {
 	e.Helper()
-	err := os.RemoveAll(e.RootPath)
-	if err != nil {
-		// In CI, Windows sometimes lags behind in marking a resource
-		// as unused. This causes otherwise passing tests to fail.
-		// So ignore errors during cleanup.
-		e.Logf("error cleaning up test directory %q: %v", e.RootPath, err)
+	for _, path := range []string{e.HomePath, e.RootPath} {
+		if err := os.RemoveAll(path); err != nil {
+			// In CI, Windows sometimes lags behind in marking a resource
+			// as unused. This causes otherwise passing tests to fail.
+			// So ignore errors during cleanup.
+			e.Logf("error cleaning up test directory %q: %v", path, err)
+		}
 	}
 }
 
-// DeleteEnvironment deletes the environment's RootPath, and everything
-// underneath it. It tolerates failing to delete the environment.
+// DeleteEnvironment deletes the environment's HomePath and RootPath, and everything
+// underneath them. It tolerates failing to delete the environment.
 func (e *Environment) DeleteEnvironmentFallible() error {
 	e.Helper()
-	return os.RemoveAll(e.RootPath)
+	err1 := os.RemoveAll(e.HomePath)
+	err2 := os.RemoveAll(e.RootPath)
+	return errors.Join(err1, err2)
 }
 
-// DeleteIfNotFailed deletes the environment's RootPath if the test hasn't failed. Otherwise
+// DeleteIfNotFailed deletes the environment's HomePath and RootPath if the test hasn't failed. Otherwise
 // keeps the files around for aiding debugging.
 func (e *Environment) DeleteIfNotFailed() {
 	if !e.T.Failed() {

